roles/metrics-export.lua

```diff
@@ -258,24 +278,52 @@ local function apply_http(conf)
     local enabled = {}
     for _, node in ipairs(conf) do
         if #(node.endpoints or {}) > 0 then
-            local host, port, err = parse_listen(node.listen)
-            if err ~= nil then
-                error("failed to parse URI: " .. err, 2)
+            local host, port, target
+            if node.server ~= nil then
+                target = {
+                    value = node.server,
+                    is_httpd_role = true,
+                }
+            elseif node.listen ~= nil then
+                local err
+                host, port, err = parse_listen(node.listen)
+                if err ~= nil then
+                    error("failed to parse URI: " .. err, 2)
+                end
+                target = {
+                    value = node.listen,
+                    is_httpd_role = false,
+                }
+            else
+                target = {
+                    value = httpd_role.DEFAULT_SERVER_NAME,
+                    is_httpd_role = true,
+                }
             end
-            local listen = node.listen
 
             http_servers = http_servers or {}
-            enabled[listen] = true
+            -- Since the 'listen' and 'server' names of other servers in the config may be
+            -- the same, we create a unique string concatenating the key name and information
+            -- about whether it is an httpd key or not.
+            enabled[target.value .. tostring(target.is_httpd_role)] = true
```

Comment:
The target.value could be a number in case listen: port . The concatenation `target.value .. tostring(target.is_httpd_role)` will fail in this case.

Reply:
I still don't get it why it worked, but added this conversion.